plugins/woocommerce-blocks/assets/js/blocks/product-collection/deprecated.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import save from './save';
+import type { ProductCollectionQuery } from './types';
+
+const v1 = {
+	attributes: {
+		...metadata.attributes,
+		query: {
+			type: 'object',
+			properties: {
+				woocommerceOnSale: {
+					type: 'boolean',
+					default: false,
+				},
+			},
+		},
+	},
+	/**
+	 * Force this deprecation to run whenever `woocommerceOnSale` has a boolean value.
+	 *
+	 * @param attributes                         The old attributes.
+	 * @param attributes.query                   The old query.
+	 * @param attributes.query.woocommerceOnSale The old woocommerceOnSale.
+	 *
+	 * @return True when the woocommerceOnSale attribute is boolean.
+	 */
+	isEligible( attributes: { query: { woocommerceOnSale: boolean } } ) {
+		const { query } = attributes;
+		return (
+			query &&
+			( query.woocommerceOnSale === true ||
+				query.woocommerceOnSale === false )
+		);
```

Comment:
nit: We can use `typeof query.woocommerceOnSale === 'boolean'` here instead.

Reply:
done -> [2a04133](https://github.com/woocommerce/woocommerce/pull/50808/commits/2a04133d9858a7ad5cacc3ca71c74ef6a4b8d21e)